helfomat-infrastructure-elasticsearch/src/main/kotlin/de/helfenkannjeder/helfomat/infrastructure/elasticsearch/organization/ElasticsearchOrganizationRepository.kt

```diff
@@ -0,0 +1,209 @@
+        )
+            .sortedBy { it.name }
+    }
+
+    override fun findGeoPointsOfOrganizationsInsideBoundingBox(position: GeoPoint?, distance: Double, boundingBox: BoundingBox): List<GeoPoint> {
+        val boolQueryBuilder = QueryBuilders.boolQuery()
+        val positionQuery = QueryBuilders.boolQuery()
+            .must(filterBox(boundingBox))
+        if (position != null)
+            positionQuery.mustNot(filterDistance(position, distance))
+        boolQueryBuilder.filter(positionQuery)
+        val organizations = search(boolQueryBuilder)
+        return extractOrganizations(emptyList(), organizations)
+            .map { obj: ScoredOrganization -> obj.organization }
+            .map { obj: Organization -> obj.defaultAddress }
+            .filter { obj: Address? -> Objects.nonNull(obj) }
+            .map { obj: Address -> obj.location }
+    }
+
+    override fun save(organizations: List<Organization>) {
+        val indexQueries = organizations
+            .map { item: Organization ->
+                IndexQueryBuilder()
+                    .withId(item.id.value)
+                    .withObject(item)
+            }
+            .map { builder: IndexQueryBuilder -> builder.withType(elasticsearchConfiguration.type.organization) }
+            .map { builder: IndexQueryBuilder -> builder.withIndexName(indexName) }
+            .map { obj: IndexQueryBuilder -> obj.build() }
+        elasticsearchTemplate.bulkIndex(indexQueries)
+    }
+
+    fun createIndex(mapping: String?) {
+        if (!elasticsearchTemplate.indexExists(indexName)) {
+            elasticsearchTemplate.createIndex(indexName)
+            elasticsearchTemplate.putMapping(indexName, elasticsearchConfiguration.type.organization, mapping)
+        }
+    }
+
+    private fun buildQueryForOrganizationWithSameTypeInDistance(defaultAddress: Address?, organizationType: OrganizationType, distanceInMeters: Long): BoolQueryBuilder {
+        val organizationListQuery = QueryBuilders.boolQuery()
+        organizationListQuery.must(QueryBuilders.termQuery("organizationType", organizationType.name))
+        if (defaultAddress == null) {
+            organizationListQuery.mustNot(QueryBuilders.existsQuery("defaultAddress"))
+        } else {
+            val locationToCheck = defaultAddress.location
+            val geoDistanceQuery = QueryBuilders.geoDistanceQuery("defaultAddress.location")
+                .point(locationToCheck.lat, locationToCheck.lon)
+                .distance(distanceInMeters.toDouble(), DistanceUnit.METERS)
+            organizationListQuery.must(geoDistanceQuery)
+        }
+        return organizationListQuery
+    }
+
+    private fun findOrganizationsWithQuestionsAndFilter(questionAnswers: List<QuestionAnswer>, filter: QueryBuilder): List<ScoredOrganization> {
+        val boolQueryBuilder = QueryBuilders.boolQuery()
+        questionAnswers
+            .filter { questionAnswer: QuestionAnswer -> Objects.nonNull(questionAnswer.answer) }
```

Comment:
it kotlin you usually don't name the lambda parameters in such a case and just use `it`. You also don't need to type them.

Reply:
In this case, the line would look like

`.filter { Objects.nonNull(it.answer) }`